packages/opencensus-web-instrumentation-zone/src/perf-resource-timing-selector.ts

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span } from '@opencensus/web-core';
+import { XhrPerformanceResourceTiming } from './zone-types';
+
+/**
+ * Get Browser's performance resource timing data associated to a XHR.
+ * Some XHRs might have two or one performance resource timings as one of them
+ * is the CORS pre-flight request and the second is related to the actual HTTP
+ * request.
+ * In overall, the algorithm to get this data takes the best fit for the span,
+ * this means the closest performance resource timing to the span start/end
+ * performance times is the returned value.
+ * @param xhrUrl
+ * @param span
+ */
+export function getXhrPerfomanceData(
```

Comment:
Would it make sense to keep a `Set` of already assigned performance entries? If so, it should be cleared when you clear the performance entry buffer. Then you could skip out on anything already assigned